### ISSUE TITLE
Dictionary of Glulx games can now contain characters outside of ISO 8859-1

### ISSUE THREAD

**Issue by DavidGriffith** [DavidGriffith/inform6lib#33]
I just noticed this item from the release notes for 6.32 of the Inform6 Compiler:

> The dictionary of Glulx games can now contain characters outside of ISO 8859-1. There is a new memory setting, $DICT_CHAR_SIZE: by default this is 1, but setting it to 4 causes the compiler to create a dictionary containing 32-bit Unicode characters.
> 
> However, this also requires library changes to be useful, as the library makes use of this dictionary table. 

I vaguely recall doing some sort of modification for this, but I'm not sure.

**Comment by vlaviano** [comment]
I've investigated this issue. The required library changes are nontrivial and have not yet been made (you'd recall having done so). Nor have they been made to I7. I have a good understanding of what needs to be done, so I've created a branch and started making these changes. However, it might not be worth delaying 6.12.1 for this enhancement.

**Comment by DavidGriffith** [comment]
I agree with your assessment.  I'll mark this for 6.12.2.

**Comment by vlaviano** [comment]
Ran across some [experimental work](http://inform7.com/extensions/Andrew%20Plotkin/Unicode%20Parser/index.html) that @erkyrath did in this area for the I7 parser. As the I6 and I7 parsers are similar, it should be a useful reference.

**Comment by erkyrath** [comment]
Yeah, it's down in the code that hasn't changed much since before I7.

Basically you change the buffer and buffer2 arrays to --> arrays, and then update any code that accesses them.
